core/development/ui/UpgradeUI.ts

```diff
@@ -0,0 +1,37 @@
+import { game } from '../Game';
+import { upgradePiece } from '../LogicAdapter';
+import { Player } from '../logic/Players';
+import { Piece } from '../logic/pieces/Piece';
+
+export function showUpgradeablePiecesElements(
+  piece: Piece,
+  upgradeablePieces: Array<new (player: Player) => Piece>,
+) {
+  const upgradesContainer = document.getElementById('piece-upgrades-container');
+  if (!upgradesContainer) return;
+
+  upgradesContainer.innerHTML = '';
+
+  upgradeablePieces.forEach((upgradeablePieceType) => {
+    const upgradeablePiece = new upgradeablePieceType(game.getCurrentPlayer());
```

Comment:
I think this should preferably not be in the UI section

Reply:
If we want to access `name` and `resource`, it needs to be